File: 0799-champagne-tower/0799-champagne-tower.py
class Solution:
    def champagneTower(self, poured: int, query_row: int, query_glass: int) -> float:
        ans = [[0] * k for k in range(1, 102)]
        ans[0][0] = poured
        for r in range(query_row + 1):
            for c in range(r + 1):
                quotient = (ans[r][c] - 1.0) / 2.0
                if quotient > 0:
                    ans[r + 1][c] += quotient
                    ans[r + 1][c + 1] += quotient
                
        return min(1, ans[query_row][query_glass])

File: 0799-champagne-tower/0799-champagne-tower.cpp
class Solution {
public:
    double champagneTower(int poured, int query_row, int query_glass) {
        vector<vector<double>> ans;
        for (int i = 1; i <= 101; ++i)
            ans.push_back(vector<double>(i, 0.0));
        
        ans[0][0] = poured;
        for (int r = 0; r <= query_row; ++r){
            for (int c = 0; c <= r; ++c){
                double quotient = (ans[r][c] - 1.0) / 2.0;
                if (quotient > 0){
                    ans[r + 1][c] += quotient;
                    ans[r + 1][c + 1] += quotient;
                }
            }
        }
        
        return min(1.0, ans[query_row][query_glass]);
    }
};